packages/ember-testing/tests/helpers_test.js

```diff
@@ -406,13 +416,22 @@ test("`wait` respects registerWaiters with optional context", function() {
     }
   };
 
+  var other=0;
+  function otherWaiter() {
+    return ++other > 2;
+  }
+
   run(App, App.advanceReadiness);
   Test.registerWaiter(obj, obj.ready);
+  Test.registerWaiter(otherWaiter);
 
   App.testHelpers.wait().then(function() {
     equal(obj.ready(), true, 'should not resolve until our waiter is ready');
     Test.unregisterWaiter(obj, obj.ready);
-    equal(Test.waiters.length, 0, 'should not leave a waiter registered');
+    equal(Test.waiters.length, 1, 'should not leave the waiter registered');
+    App.testHelpers.wait().then(function() {
```

Comment:
This nesting seems odd. Can you return wait() here, and then chain the past assertion off the outside promise (instead of nesting another level)?